### ISSUE TITLE
model.eval() does not reach internal layers

### ISSUE THREAD

**Issue by vikcas** [LiamMaclean216/Pytorch-Transfomer#6]
Dear all, 

I've been trying your code. I've adapted it to my data and I've trained a model. But then, at inference if the random seed is not fix the model produces different results each time.
I think that the model.eval() method that should change the behavior of the layers that perform mini-batch normalization (inside the encoder, for instace) is not reaching that part of the model.

Best.
